src/scripts/domParsers/videoExtractor.ts

```diff
@@ -0,0 +1,9 @@
+export interface VideoExtractor {
+	// Our 3 Domains all have the concept of a "VideoId" embedded somewhere in the page
+	getVideoIds(pageUrl: string, pageContent: string): string[];
+
+	// We can use the video ideas to generate or retrieve src values that we can embed into our article preview
+	getVideoSrcValues(pageUrl: string, pageContent: string): string[];
```

Comment:
I considered passing in `PageInfo`. I will investigate if it easy enough to do. It would simplify the interface and make it a little more flexible.